.github/workflows/pr-release.yml

```diff
@@ -0,0 +1,65 @@
+# Push a release to the lean4-pr-releases repository, whenever someone pushes to a PR branch.
+
+# This needs to run with the `secrets.PR_RELEASES_TOKEN` token available,
+# but PR branches will generally come from forks,
+# so it is not possible to run this using the `pull_request` or `pull_request_target` workflows.
+# Instead we use `workflow_run`, which essentially allows us to escalate privileges
+# (but only runs the CI as described in the `master` branch, not in the PR branch).
+
+name: PR release
+
+on:
+  workflow_run: # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run
+    workflows: [CI]
+    types: [completed]
+
+jobs:
+  on-success:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: Retrieve information about the original workflow
+        uses: potiuk/get-workflow-origin@v1_1 # https://github.com/marketplace/actions/get-workflow-origin
+        id: workflow-info
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }} d
+          sourceRunId: ${{ github.event.workflow_run.id }}
+      - name: Checkout
+        # Only proceed if the previous workflow had a pull request number.
+        if: ${{ steps.workflow-info.outputs.pullRequestNumber != '' }}
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.PR_RELEASES_TOKEN }}
+          # We only need to checkout for tagging, so depth 0 is fine:
```

Comment:
Ah, indeed, I misunderstood what was happening in `ci.yml`. I guess in `ci.yml` we have `fetch-depth: 0` in order to retrieve more-than-the-usual information so that the "Prepare Nightly Release" step can prepare the `diff.md`.

I will change this to depth 1, instead!

Looking at [the logs](https://github.com/semorrison/lean4/actions/runs/5950417117/job/16138168423#step:3:763) from running a PR against this setup on my fork, this is just checking out `master`, not the PR.

I think that's fine, as we don't interact with the local checkout at all except to create a tag and push it to the `lean4-pr-releases` repo. Nevertheless I'll add a comment noting this to avoid confusion later!

Reply:
> Looking at [the logs](https://github.com/semorrison/lean4/actions/runs/5950417117/job/16138168423#step:3:763) from running a PR against this setup on my fork, this is just checking out `master`, not the PR.

That's what I expected as well, but then doesn't that mean we tag the wrong commit?

Reply:
Oops.